### PR TITLE
[To rel/1.1][IOTDB-5738] Retain 2 copies of snapshot

### PR DESCRIPTION
**Pull request by SzyWilliam**
cherry pick of https://github.com/apache/iotdb/pull/9474

### CODE REVIEW

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/SnapshotStorage.java [hunk 1]
```diff
@@ -166,11 +167,14 @@ public void format() throws IOException {}
...
-    if (sortedSnapshotDirs == null || sortedSnapshotDirs.length == 0) {
+    final Path[] sortedSnapshotDirs = getSortedSnapshotDirPaths();
+    if (ArrayUtils.isEmpty(sortedSnapshotDirs)) {
       return;
     }
-    for (int i = 0; i < sortedSnapshotDirs.length - 1; i++) {
+
+    final int retainIndex =
+        snapshotRetentionPolicy.getNumSnapshotsRetained() - 1; // snapshot rank starts from 0
+    for (int i = retainIndex; i < sortedSnapshotDirs.length - 1; i++) {
```
Comment (wangchao316) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/SnapshotStorage.java:177]: The modification code and master branch are different. why?

Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/SnapshotStorage.java]: Fixed. Now they are the same. Thanks!

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**